xgcm/test/test_metrics.py

```diff
@@ -138,6 +136,105 @@ def test_get_metric(axes, data_var, drop_vars, metric_expected_list, expected_er
+            ("X", "Y"),
+            "v",
+            "area_e",
+            {("X", "Y"): "area_e", "X": "dx_n", "Y": "dy_n"},
+            "cond2",
+        ),  # Condition 2, case b: get_metric should select for the metric with matching axes and interpolate from there,
+        # even if other metrics in the desired positions are available
+        (
+            "set_metrics",
+            ("X", "Y"),
+            "v",
+            None,
+            {"X": "dx_n", "Y": "dy_n"},
+            "cond3",
+        ),  # Condition 3: use provided metrics to calculate for required metric
+        (
+            "set_metrics",
+            ("X", "Y"),
+            "v",
+            None,
+            {"X": "dx_t", "Y": "dy_n"},
+            "cond4a",
+        ),  # Condition 4, case a: 1 metric on the wrong position (must interpolate before multiplying)
+        (
+            "set_metrics",
+            ("X", "Y"),
+            "v",
+            None,
+            {"X": "dx_t", "Y": "dy_t"},
+            "cond4b",
+        ),  # Condition 4, case b: 2 metrics in the wrong position (must interpolate before multiplying)
+        (
+            "set_metrics",
+            ["X", "Y", "Z"],
+            "tracer",
+            None,
+            {("X", "Y"): "area_t", "Z": "dz_t"},
+            "cond3",
+        ),  # Adapt from original version of test_get_metric
+    ],
+)
+def test_get_metric_with_conditions(
+    set_metrics, metric_axes, ds_array, ds_expected, set_metric_vars, expected_metric
+):
+
+    ds, coords, metrics = datasets_grid_metric("C")
+    if set_metrics == "metrics":
+        grid = Grid(ds, coords=coords, metrics=metrics)
+    else:
+        grid = Grid(ds, coords=coords)
+        for k, v in set_metric_vars.items():
+            grid.set_metrics(k, v)
+
+    get_metric = grid.get_metric(ds[ds_array], metric_axes)
+
+    if expected_metric == "cond1":
+        expected_metric = ds[ds_expected].reset_coords(drop=True)
+    elif expected_metric == "cond2":
+        expected_metric = grid.interp(ds[ds_expected], ("X", "Y"))
+    elif expected_metric == "cond3":
+        metric_values = [ds[v] for k, v in set_metric_vars.items()]
+        expected_metric = (
+            functools.reduce(operator.mul, metric_values, 1)
+        ).reset_coords(drop=True)
+    elif expected_metric == "cond4a":
+        metric_values = [ds[v] for k, v in set_metric_vars.items()]
+        metric_interp = grid.interp(metric_values[0], "Y")
+        expected_metric = (metric_interp * metric_values[1]).reset_coords(drop=True)
+    elif expected_metric == "cond4b":
+        metric_values = [ds[v] for k, v in set_metric_vars.items()]
+        metric_interp_1 = grid.interp(metric_values[0], "Y")
+        metric_interp_2 = grid.interp(metric_values[1], "Y")
+        expected_metric = (metric_interp_1 * metric_interp_2).reset_coords(drop=True)
+
+    xr.testing.assert_equal(get_metric, expected_metric)
+    xr.testing.assert_allclose(get_metric, expected_metric)
```

Comment:
Only one of these is needed: the only difference is whether they use `np.testing.array_equal` or `np.testing.array_allclose` under the hood. The latter allows you to ignore minor floating point differences. If it passes with `equal` use that, if not use `allclose`